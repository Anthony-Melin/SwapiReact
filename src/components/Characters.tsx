import ListValueFactory from './__components__/ListValueFactory.tsx'

export default ListValueFactory('characters', 'Characters', 'characters')
