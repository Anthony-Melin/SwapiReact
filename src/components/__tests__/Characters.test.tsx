import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Characters from '../Characters.tsx'

describe('Characters', () => {
    describe('with empty list', () => {
        beforeEach(() => {
            render(<Characters values={[]} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('characters')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('characters')).toHaveTextContent('Characters')
        })

        it('should have empty list label', () => {
            expect(screen.getByTestId('characters-empty')).toHaveTextContent('No characters linked')
        })
    })

    describe('with filled list', () => {
        beforeEach(() => {
            render(<Characters values={['VALUE_1', 'VALUE_2']} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('characters')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('characters')).toHaveTextContent('Characters')
        })

        it('should contains elements', () => {
            expect(screen.getByTestId('characters-0')).toHaveTextContent('VALUE_1')
            expect(screen.getByTestId('characters-1')).toHaveTextContent('VALUE_2')
        })
    })
})
