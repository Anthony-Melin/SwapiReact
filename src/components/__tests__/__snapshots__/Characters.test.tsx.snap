// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Characters > with empty list > should match snapshot 1`] = `
<div
  data-testid="characters"
>
  <p
    data-testid="characters-title"
  >
    Characters
  </p>
  <p
    data-testid="characters-empty"
  >
    No 
    characters
     linked
  </p>
</div>
`;

exports[`Characters > with filled list > should match snapshot 1`] = `
<div
  data-testid="characters"
>
  <p
    data-testid="characters-title"
  >
    Characters
  </p>
  <ul>
    <li
      data-testid="characters-0"
    >
      VALUE_1
    </li>
    <li
      data-testid="characters-1"
    >
      VALUE_2
    </li>
  </ul>
</div>
`;
